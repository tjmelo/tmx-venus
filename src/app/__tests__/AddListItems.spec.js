import toAddList from '../modules/AddListItems';

document.body.innerHTML = '<button id="button"/>';
const testButtonAddToList = document.querySelector('#button');

describe('Should render function to Add new item in list', () => {
  it('Should render call function with reference button', () => {
    const testToAddList = jest.fn();

    testToAddList.mockImplementation((argTest) => toAddList(argTest));

    // do
    testToAddList(testButtonAddToList);

    // then
    expect(testToAddList).toHaveReturned();
    expect(testToAddList).toHaveBeenCalledWith(testButtonAddToList);
    expect(testToAddList).toHaveBeenCalledTimes(1);

    // Expects fidelity with scope of function
    expect(testToAddList).toMatchSnapshot();
  });
});
