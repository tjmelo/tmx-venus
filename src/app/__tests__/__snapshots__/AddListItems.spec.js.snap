// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render function to Add new item in list Should render call function with reference button 1`] = `
[MockFunction] {
  "calls": [
    [
      <button
        id="button"
      />,
    ],
  ],
  "results": [
    {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;
